SPEC

```diff
@@ -113,6 +113,25 @@ be implemented by the server.
                         fatal(message, &block)
 <tt>rack.multipart.buffer_size</tt>:: An Integer hint to the multipart parser as to what chunk size to use for reads and writes.
 <tt>rack.multipart.tempfile_factory</tt>:: An object responding to #call with two arguments, the filename and content_type given for the multipart form field, and returning an IO-like object that responds to #<< and optionally #rewind. This factory will be used to instantiate the tempfile for each multipart form file upload field, rather than the default class of Tempfile.
+
+Servers that choose to support WebSockets and SSE connections should follow these additional environment specifications:
+<tt>rack.upgrade?</tt>:: Is nil or missing if no connection upgrade is possible,
+                         :websocket for a WebSocket upgrade, or :sse for an SSE upgrade.
+<tt>rack.upgrade</tt>:: Is used to pass a handler instance or class back to the server for WebSocket and SSE connections.
```

Comment:
Maybe, now that we support static handlers, this line should be altered?

Perhaps:

> <tt>rack.upgrade</tt>:: Is used to pass a handler object back to the server for WebSocket and SSE connections.

Reply:
Sure, minor change.